src/simplify_algebra.cpp

```diff
@@ -985,20 +985,34 @@ struct find_split_reshape
         auto rsp_lens    = rsp->get_shape().lens();
         auto rsp_strides = rsp->get_shape().strides();
         rsp_strides.insert(rsp_strides.begin(), rsp_strides[0] * rsp_lens[0]);
-        auto ait = std::find(rsp_strides.begin(), rsp_strides.end(), slc_dim_size);
+
+        auto ait     = std::find(rsp_strides.begin(), rsp_strides.end(), slc_dim_size);
+        int rsp_axis = -1;
         if(ait == rsp_strides.end())
         {
             return;
         }
-        int rsp_axis = std::distance(rsp_strides.begin(), ait);
-
+        else if(ait == rsp_strides.end() - 1)
+        {
+            // edge case
+            // slice_dim == 1, in that case it could match with last stride of 1.
+            // it should accumulate lengths from last dim in that case. discount 1 to avoid going
+            // out of bounds.
```

Comment:
Where would we go out of bounds?

Reply:
e.g.  slice output of lens `[1024, 128, 1]`

reshape output shape = `[1024, 128][128, 1]`
`slice_dim_size = 1`, 
`auto rsp_strides = rsp->get_shape().strides();`
`rsp_strides.insert(rsp_strides.begin(), rsp_strides[0] * rsp_lens[0]);` --> `rsp_strides = [1024 * 128, 128, 1]`
` auto ait     = std::find(rsp_strides.begin(), rsp_strides.end(), slc_dim_size);` ---> `ait = 2`. 
`rsp_axis = std::distance(rsp_strides.begin(), ait);` ---> `rsp_axis = 2`. 

Now, inside the following, it would go out of bounds since `lens()` would only be of length 2. 
`std::transform(vec_rsp.begin(), vec_rsp.end(), vec_dims.begin(), [&](auto is) {
            return is->get_shape().lens()[rsp_axis];
        });`